esp/src/src-react/components/controls/Grid.tsx

```diff
@@ -237,12 +237,21 @@ const FluentStoreGrid: React.FunctionComponent<FluentStoreGridProps> = ({
         });
     });
 
+    const abortController = React.useRef<AbortController>();
+
+    useDeepEffect(() => {
+        if (abortController.current) {
+            abortController.current.abort({ message: "Grid aborting stale request" });
+        }
+        abortController.current = new AbortController();
+    }, [abortController], [query]);
```

Comment:
Why is a DeepEffect needed?  Wouldn't a useEffect triggered by a query change suffice?